Public Release for Fall24 meta-release

bigludo7 opened the pull request:
#### What type of PR is this?

Add one of the following kinds:

* subproject management



#### What this PR does / why we need it:

Related to issue #107 

#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #


#### Special notes for reviewers:

Missing link for Test Statement
PR for Test Definition no yet merged

#### Changelog input

```
 Public release number_verification v1.0.0

```

#### Additional documentation 

This section can be blank.



```
docs

```

hdamker commented on CHANGELOG.md:
```diff
@@ -3,15 +3,66 @@
+
+## number_verification 1.0.0
+
+**number_verification 1.0.0 is the public release for v1.0.0 of the NumberVerification API.**
```
How about:
```suggestion
**number_verification 1.0.0 is the release of the first stable version of the NumberVerification API.**
```

hdamker commented on documentation/API_documentation/NumberVerification-Readiness-Checklist.md:
```diff
@@ -9,14 +9,13 @@ Checklist for number_verification 1.0.0-rc.1 in release r1.1
-|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   tbd   | link |
-
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y   | [get](/documentation/API_documentation/NumberVerification_device_phone_number_User_Story.md) [verify](/documentation/API_documentation/NumberVerification_verify_User_Story.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [PR #124](https://github.com/camaraproject/NumberVerification/pull/124/files) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   Y  | [PR #124](https://github.com/camaraproject/NumberVerification/pull/124/files) |
```
The status is either "tbd" with the PR, or you consider that PR 124 must be merged before the release PR, than you can already link here the coming .feature file.

bigludo7 replied:
We have just merged the Test Definition.
I've updated the rc1.2 branch accordingly.

hdamker reviewed (commented):
LGTM, one suggestion and one comment.

Please add @camaraproject/release-management_maintainers, as it is ready for review.

AxelNennker commented:
Please review my ICM review on 1.1
https://github.com/camaraproject/IdentityAndConsentManagement/issues/189#issuecomment-2314746067

rartych commented:
The link in the last line of README.md is wrong.
It should go to https://wiki.camaraproject.org/display/CAM/NumberVerification  or  https://wiki.camaraproject.org/x/jQOeAQ

hdamker commented:
@bigludo7 please consider to fix https://github.com/camaraproject/NumberVerification/issues/142 as part of this PR. Then all API YAMLs across the meta-release will have the same format.

bigludo7 commented:
> @bigludo7 please consider to fix #142 as part of this PR. Then all API YAMLs across the meta-release will have the same file name format.

Done :)

hdamker commented on code/Test_Definitions/NumberVerification_device_phone_number_share.feature:
```diff
@@ -5,7 +5,7 @@ Feature: Camara Number Verification API device phone number share
 # Input to be provided by the implementation to the tests
 # References to OAS spec schemas refer to schemas specified in
-# https://raw.githubusercontent.com/camaraproject/NumberVerification/main/code/API_definitions/number_verification.yaml
+# https://raw.githubusercontent.com/camaraproject/NumberVerification/main/code/API_definitions/number-verification.yaml
```
Is somewhere defined which version of the API definition the .feature file refers to? This link goes to main

Maybe a relative link would be better to stay within the tag/branch, but I don't this is possibly for raw content. An alternative would be to link to `/code/API_definitions/number_verification.yaml`

bigludo7 replied:
I've applied your suggestion.

hdamker commented on documentation/API_documentation/NumberVerification-Readiness-Checklist.md:
```diff
@@ -1,22 +1,21 @@
-
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | Embed documentation into API spec - [link](/code/API_definitions/number-verification.yaml)  |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y   | [get](/documentation/API_documentation/NumberVerification_device_phone_number_User_Story.md) [verify](/documentation/API_documentation/NumberVerification_verify_User_Story.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [get](/code/Test_Definitions/NumberVerification_device_phone_number_share.feature) [verify](/code/Test_Definitions/NumberVerification_verify.feature) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   Y  | [get](/code/Test_Definitions/NumberVerification_device_phone_number_share.feature) [verify](/code/Test_Definitions/NumberVerification_verify.feature) |
```
```suggestion
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [get](code/Test_Definitions/number-verification-device-phone-number-share.feature) [verify](/code/Test_Definitions/number-verification-verify.feature) |
|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   Y  | [get](code/Test_Definitions/number-verification-device-phone-number-share.feature) [verify](code/Test_Definitions/number-verification-verify.feature) |

```

bigludo7 commented:
> In the path: section there is `X-Correlator`, the guideline was changed to use `x-correlator` (lower case) https://github.com/camaraproject/Commonalities/pull/194/files

Thanks to catch this one Rafal
Fixed.

jlurien commented on documentation/API_documentation/NumberVerification-Readiness-Checklist.md:
```diff
@@ -1,22 +1,22 @@
-|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   tbd   | [PR #124](https://github.com/camaraproject/NumberVerification/pull/124/files) |
-|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   tbd  | [PR #124](https://github.com/camaraproject/NumberVerification/pull/124/files) |
-|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   tbd   | link |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | Embed documentation into API spec - [link](/code/API_definitions/number-verification.yaml)  |
```
```suggestion
|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | Embedded documentation into API spec - [link](/code/API_definitions/number-verification.yaml)  |

```

jlurien reviewed (commented):
- Update versions in test plans:  Given the resource "/number-verification/v0 ->  Given the resource "/number-verification/v1

bigludo7 commented:
> * Update versions in test plans:  Given the resource "/number-verification/v0 ->  Given the resource "/number-verification/v1

Thank @jlurien 
fixed

jlurien commented:
Let's wait to decision on https://github.com/camaraproject/ReleaseManagement/issues/89 to merge this

bigludo7 commented:
Following discussion in today WG meeting & sync with @hdamker I've updated the yaml to add a clarification before the sequence diagram. Changelog has been also updated.
@hdamker @fernandopradocabrillo @jlurien I will need your approval (I understood @AxelNennker is off this week).
Thanks.

hdamker commented on code/API_definitions/number-verification.yaml:
```diff
@@ -30,6 +30,8 @@ info:
     # Sequence Diagram
     Number Verification API uses the **standard [OIDC Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth)**. The following diagram will help to clarify the end-to-end process, including previous steps prior to this API call.
 
+    Note that the diagram shows just an example of a direct integration from developer's application and MNO's authserver and API, other scenarios of indirect integration via Aggregator/Channel Partner should also be considered. Specific authorization flows may apply, as defined by e.g. GSMA OGW
```
```suggestion
    Note that the diagram shows just an example of a direct integration from developer's application and MNO's authserver and API, other scenarios of indirect integration via Aggregator/Channel Partner should also be considered. Specific authorization flows may apply, as defined by e.g. GSMA Open Gateway (OGW).
```

hdamker replied:
Avoid abbreviations where possible.

tanjadegroot commented:
Please see https://github.com/camaraproject/ReleaseManagement/issues/89 on how to update the Test result statement in the API readiness checklist for this Fall24 meta-release.

bigludo7 commented:
> Please see [camaraproject/ReleaseManagement#89](https://github.com/camaraproject/ReleaseManagement/issues/89) on how to update the Test result statement in the API readiness checklist for this Fall24 meta-release.

Thanks @tanjadegroot 
Applied
cc @hdamker

jlurien reviewed (approved):
Release PR approved (on behalf of Release Management)

@bigludo7 PR ready to be merged and r1.2 can be created.Please update https://wiki.camaraproject.org/display/CAM/NumberVerification++API+Release+Tracking with date for M4 and link to r1.2 when it is created. Thanks

fernandopradocabrillo reviewed (approved):
LGTM
